shared/elements/gaia_grid/js/items/mozapp.js

```diff
@@ -184,6 +193,19 @@
       return localized || this.descriptor.short_name || this.descriptor.name;
     },
 
+    asyncName: function() {
+      var userLang = document.documentElement.lang;
+
+      var ep = this.entryPoint || undefined;
+
+      return this.app.getLocalizedValue('short_name', userLang, ep).then(
+        short_name => localizeString(short_name),
```

Comment:
nit: I would opt for ```shortName``` instead of short_name as this is a variable and not an object property.